profiling/report.py

```diff
@@ -39,37 +49,51 @@
 # Read the profiling results
 sio = StringIO()
 p = pstats.Stats(str(prof_path), stream=sio)
-p.print_stats(str(root).replace("\\", "\\\\"))
-# p.print_callers(str(root))
+p.print_stats("pyrealm")
 report = sio.getvalue()
+print(report)
 
 # Convert to a DataFrame and save to CSV
-report = report.replace(str(root / "*")[:-1], "")  # remove common path
 _, report = re.split(r"\n(?= +ncalls)", report, 1)  # remove header
 df = (
-    pd.read_csv(StringIO(report), sep=" +", engine="python")
+    pd.read_csv(StringIO(report), sep=" +", engine="python", on_bad_lines="skip")
     .rename(columns={"percall": "tottime_percall", "percall.1": "cumtime_percall"})
     .sort_values(by="cumtime_percall", ascending=False)
     .query("cumtime > 0.01")
 )
-df[["filename", "lineno", "function"]] = df.pop(
-    "filename:lineno(function)"
-).str.extract(r"(.*?):(.*?)\((.*)\)", expand=True)
+s = df.pop("filename:lineno(function)")
+df[["filename", "lineno", "function"]] = s.str.extract(
+    r"(.*):(.*?)\((.*)\)", expand=True
+)
+df.dropna(inplace=True)
+if args.selector:
+    df = df[df.filename.str.contains(args.selector)]
+i = next(
+    i
+    for i in range(99)
+    if (lambda ss: any(s[i] != ss[0][i] for s in ss))(df.filename.iloc)
+)
+df["filename"] = [s[i:] for s in df.filename]  # remove common prefix
```

Comment:
What is this doing? And it looks like a lambda is being used as an anonymous function? This is not easy to follow.

Reply:
`i` is the maximum index such that `s[i]` is the same for every filename `s`. Therefore these lines are used to remove the common prefix of all the filenames in `df`.

Reply:
Can you give a concrete example of the data is being changed here?

Reply:
These are several filenames before the change:
```
D:\Personal\Projects\pyrealm\pyrealm\pmodel\pmodel.py
D:\Personal\Projects\pyrealm\tests\profiling\test_profiling_pmodel.py
D:\Personal\Projects\pyrealm\pyrealm\pmodel\pmodel.py
D:\Personal\Projects\pyrealm\pyrealm\pmodel\subdaily.py
D:\Personal\Projects\pyrealm\pyrealm\pmodel\jmax_limitation.py
D:\Personal\Projects\pyrealm\pyrealm\pmodel\jmax_limitation.py
D:\Personal\Projects\pyrealm\pyrealm\splash\splash.py
D:\Personal\Projects\pyrealm\tests\profiling\test_profiling_splash.py
```

And after the change:
```
pyrealm\pmodel\pmodel.py
tests\profiling\test_profiling_pmodel.py
pyrealm\pmodel\pmodel.py
pyrealm\pmodel\subdaily.py
pyrealm\pmodel\jmax_limitation.py
pyrealm\pmodel\jmax_limitation.py
pyrealm\splash\splash.py
tests\profiling\test_profiling_splash.py
 ```

Reply:
Seems like it would be easier and faster to get the shared path through `__file__` rather than this!

Reply:
I did this initially but after I installed the package using `pip install -e .`, the files are located somewhere in the `site-packages` folder of the python environment...

Reply:
But that must still be discoverable from `Path` objects rather than this.

Reply:
Yeah I agree

Reply:
This is resolved as no longer part of this PR.